client/verta/verta/_internal_utils/model_dependencies.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+import ast
+import inspect
+import textwrap
+from types import ModuleType
+from typing import Callable, get_type_hints, List, Tuple, Type
+
+from verta.registry import VertaModelBase
+
+
+def list_class_functions(model_class: Type[VertaModelBase]) -> List[Tuple[str, Callable]]:
+    """List all the functions present in the provided class object."""
+    return inspect.getmembers(model_class, predicate=inspect.isfunction)
+
+
+def unwrap(func: Callable) -> Callable:
+    """Unwraps a function to get the original function object in a non-blocking manner."""
+    try:
+        return inspect.unwrap(func)  # returns last object in chain
+    except ValueError:  # if a cycle is encountered
+        return func
+
+
+def list_modules_in_function_definition(func: Callable) -> List[ModuleType]:
```

Comment:
I think it might be more accurate to call this

```suggestion
def list_modules_in_function_body(func: Callable) -> List[ModuleType]:
```

since a signature is technically also part of a function definition.

Reply:
Updated `definition` >> `body`